### ISSUE TITLE
Shard moves fail when a partitioned table has a foreign key to reference table

### ISSUE THREAD

**Issue by marcocitus** [citusdata/citus#6279]
Because of the optimization to skip validation on the target in `GetForeignConstraintCommandsToReferenceTable`, we get the following error:
```sql
drop table if exists ref;
create table ref (id int primary key);
select create_reference_table('ref');
insert into ref select s from generate_series(0,9) s;
drop table if exists test_1, test_2, test;
create table test (x int, y int references ref (id), t timestamptz default now()) partition by range (t);
create table test_1 partition of test for values from ('2022-01-01') to ('2022-12-31');
create table test_2 partition of test for values from ('2023-01-01') to ('2023-12-31'); 
insert into test (x,y) select s,s%10 from generate_series(1,100) s;
select create_distributed_table_concurrently('test','x');

select citus_move_shard_placement(102305, 'localhost', 1401, 'localhost', 1402, shard_transfer_mode := 'force_logical');
NOTICE:  issuing SELECT worker_apply_inter_shard_ddl_command (102305, 'public', 102270, 'public', 'ALTER TABLE public.test ADD CONSTRAINT test_y_fkey FOREIGN KEY (y) REFERENCES public.ref(id) NOT VALID')
ERROR:  cannot add NOT VALID foreign key on partitioned table "test_102305" referencing relation "ref_102270"
DETAIL:  This feature is not yet supported on partitioned tables.
```

**Comment by onderkalaci** [comment]
This really surprises me how not hit by users for such a long time, see https://github.com/citusdata/citus-enterprise/issues/284

**Comment by marcocitus** [comment]
Fixed by #6281
